staging/src/k8s.io/cloud-provider/controllers/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -196,7 +196,50 @@ func (c *CloudNodeLifecycleController) MonitorNodes(ctx context.Context) {
 	}
 }
 
+// getProviderID returns the provider ID for the node. If Node CR has no provider ID,
+// it will be the one from the cloud provider.
+func getProviderID(ctx context.Context, cloud cloudprovider.Interface, node *v1.Node) (string, error) {
+	var providerID string
+	var err error
```

Comment:
You don't have to initialize the var `err`

Reply:
Correct me where I am wrong but if I don't declare it here then I need to write
```
providerID, err := instances.InstanceID(ctx, types.NodeName(node.Name))
```
what is going to shadow providerID and subsequently ignore the result of
```
instances.InstanceID(ctx, types.NodeName(node.Name))
```
when used in
```
return cloud.ProviderName() + "://" + providerID, nil
```

Reply:
You are totally right, I have missed the `providerID` var 🤦🏻